### ISSUE TITLE
Server overload when clicking update button from admin page. 9.0.1 -> 9.0.4

### ISSUE THREAD

**Issue by JKomrokian** [owncloud/core#25657]
### Steps to reproduce
1.Clean install 9.0.1 and prompted to update to 9.0.4
2.Click the update button from the admin page

### Expected behaviour
The update process should start 

### Actual behaviour
The server is overloaded and connection in not possible.
The server was running 1700+ instances of occ, I was able to connect again by rebooting the server 

11377 ebfile 20 0 184m 5044 372 S 0.6 0.1 0:00.50 php /home/ebfile/public_html/occ --no-warnings config:list --private --output json

### Server configuration
**Operating system**:

**Web server:**

**Database:**
mySQL
**PHP version:**
5.5.16
**ownCloud version:** (see ownCloud admin page)
9.0.1
**Updated from an older ownCloud or fresh install:**
Fresh install
**Where did you install ownCloud from:**
Web installer
**Signing status (ownCloud 9.0 and above):**
```
No errors have been found.
```

**List of activated apps:**

```

```

**The content of config/config.php:**

```
<?php
$CONFIG = array (
  'instanceid' => '--------------',
  'passwordsalt' => '-------------------',
  'secret' => '------------------------',
  'trusted_domains' => 
  array (
    0 => '-------------------',
  ),
  'datadirectory' => '/home/ebfile/public_html/data',
  'overwrite.cli.url' => 'http://------------------',
  'dbtype' => 'mysql',
  'version' => '9.0.1.3',
  'dbname' => '------------',
  'dbhost' => 'localhost',
  'dbtableprefix' => '-----',
  'dbuser' => '---------------',
  'dbpassword' => ------------',
  'logtimezone' => 'UTC',
  'installed' => true,
  'loglevel' => 0,
  'updater.secret' => '--------------------',
);
```

**Are you using external storage, if yes which one:** local/smb/sftp/...
no
**Are you using encryption:** yes/no
no
**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
webdav to transfer large files 

```

```

### Client configuration
**Browser:**
Chrome
**Operating system:**
Windows 10
#### Web server error log
```

```

#### ownCloud log (data/owncloud.log)
```

```

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log 
c) ...
```

**Comment by saint42** [comment]
I have exactly the same error. 
Got over 3,000 processes running occ --no-warnings config:list --private --output json

**Comment by PVince81** [comment]
@VicDeo please have a look

Possibly some kind of loop that retries the same command over and over again ?

**Comment by VicDeo** [comment]
@saint42 which webserver do you use?
please attach `update.log`

**Comment by saint42** [comment]
hi - i use apache. 
My 2 OwnCloud installations had the exact same issue on two servers. 
I had to do a manual upgrade (moved the data folder, installed new version, restored config.php, did the database upgrade, and restored the data server)... I won't be of any help since I did not keep any logs.. :/ Sorry.

**Comment by VicDeo** [comment]
@JKomrokian  and what about you?
which webserver do you use/ do you have  update.log?

**Comment by VicDeo** [comment]
The most obvious reason for this is a redirect loop on the server side.

**Comment by VicDeo** [comment]
@saint42  Does it happen too when you  open http://your_domain.tld/updater in browser in your current 9.0.4?

**Comment by saint42** [comment]
@VicDeo could you explain how to setup the updater.secret ? 

updater.secret is undefined in config/config.php. Either browse the admin settings in your ownCloud and click "Open updater" or define a strong secret using
php -r 'echo password_hash("MyStrongSecretDoUseYourOwn!", PASSWORD_DEFAULT)."\n";'
and set this in the config.php.

without breaking my prod server :)
thanks

**Comment by VicDeo** [comment]
@saint42 
`php -r 'echo password_hash("SomePassword", PASSWORD_DEFAULT)."\n";'`
will produce a hash, let's say SomeHash

to be able to login using SomePassword
 you need to put this hash into your config php:
`'updater.secret'  => 'SomeHash',`

**Comment by VicDeo** [comment]
@saint42 any news?

**Comment by saint42** [comment]
 Initializing


 
 [UnexpectedValueException] 
 Could not parse a response for config:list. Please check if the current shell user can run occ command. Raw output: 
 Not allowed 
 

I logged in OwnCloud first to make sure the cookies knew me as an admin, then ran the updater.

**Comment by VicDeo** [comment]
@saint42 please check https://github.com/owncloud/core/issues/25973#issuecomment-244795782

**Comment by saint42** [comment]
ok, one step further now with the comment above:

 [UnexpectedValueException] 
 Could not parse a response for config:list. Please check if the current shell user can run occ command. Raw output: 
 Web executor is not allowed to run from a different host 
 

**Comment by PVince81** [comment]
>  Web executor is not allowed to run from a different host

This will be fixed in 9.0.6 and patches will be provided.

@VicDeo can you link with the matching ticket ? (there are so many I lost track)
Thanks.

**Comment by PVince81** [comment]
Ah, let's take this one here: https://github.com/owncloud/updater/issues/397

Closing
